examples/snake-cli/snake.rs

```diff
@@ -0,0 +1,260 @@
+use std::collections::LinkedList;
+
+use crate::coordinate::Coordinate;
+use crate::defs::RESOLUTION;
+use crate::direction::Direction;
+use neat_gru::neural_network::nn::NeuralNetwork;
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct Block {
+    cord: Coordinate,
+}
+
+impl Block {}
+#[derive(Clone, PartialEq, Debug)]
+pub struct Snake {
+    blocks: LinkedList<Block>,
+    moving_direction: Direction,
+    dir_changed: bool,
+    net: Option<NeuralNetwork<f64>>,
+}
+
+impl Snake {
+    pub fn new(net: NeuralNetwork<f64>) -> Self {
+        let mut blocks = LinkedList::new();
```

Comment:
I usually try to avoid `LinkedList` since they are very inefficient for the cache, but it is very elegant here